src/main/java/com/google/devtools/build/remote/client/LogParserUtils.java

```diff
@@ -220,6 +225,26 @@ static void printLogEntry(LogEntry entry, PrintWriter out) throws IOException {
     }
   }
 
+  private String protobufToJsonEntry(LogEntry input) {
+      String jsonString = "";
+      if (input == null) {
+          throw new RuntimeException("No input provided for parsing");
+      } else {
+          try {
+              jsonString = JsonFormat.printer()
+              .usingTypeRegistry(
+                  JsonFormat.TypeRegistry.newBuilder()
+                      .add(ExecuteOperationMetadata.getDescriptor())
+                      .build())
+                      .print(input);
+          } catch (Exception e) {
+              throw new RuntimeException("Error deserializing protobuf to json", e);
+          }
+      }
+      // We want each entry to have it's own comma
+      return jsonString;
```

Comment:
This is an unnecessarily complicated sequence of logic.
If you throw in an if block unconditionally, you don't need an else.
If your only path to return a value unconditionally is just a store for that value, you can just return the invocation.
Don't catch Exception here, be explicit and match with the (I assume) IOException, or better yet, indicate that this exception throws that one (and now you can throw the checked IOException on null input).
Better still, you can use `.print(checkNotNull(input))` from the guava Preconditions suite, since passing null here should be a bug as indicated by your RuntimeException.

Reply:
Got it, thanks